File: exercises/day13-Keyboard-First-Refactoring/kotlin/src/main/kotlin/sleightrip/elf/ElfSession.kt
package sleightrip.elf

import sleightrip.exception.CollaboratorCallException

class ElfSession private constructor() {
    
    companion object {
        private val instance = ElfSession()
        
        fun getInstance(): ElfSession = instance
    }

    fun getLoggedUser(): Elf {
        throw CollaboratorCallException(
            "ElfSession is a static collaborator and should not be used directly in unit tests"
        )
    }
}

File: exercises/day13-Keyboard-First-Refactoring/typescript/src/elf/ElfSession.ts
import { Elf } from './Elf';
import { CollaboratorCallException } from '../exception/Exceptions';

export class ElfSession {
    private static instance = new ElfSession();

    private constructor() {}

    static getInstance(): ElfSession {
        return ElfSession.instance;
    }

    getLoggedUser(): Elf {
        throw new CollaboratorCallException(
            'ElfSession is a static collaborator and should not be used directly in unit tests'
        );
    }
}
